pkg/processor/worker/worker.go

```diff
@@ -160,25 +156,19 @@ func (w *Worker) Terminate() error {
 }
 
 func (w *Worker) Drain() error {
-	w.drainedLock.Lock()
-	defer w.drainedLock.Unlock()
-
-	if !w.isDrained.Load() {
-		err := w.runtime.Drain()
-		if err == nil {
-			w.logger.DebugWith("Successfully drained worker", "workerIndex", w.index)
-			w.isDrained.Store(true)
-		}
+	if err := w.runtime.Drain(); err != nil {
 		return err
 	}
+	w.logger.DebugWith("Successfully drained worker", "workerIndex", w.index)
 	return nil
 }
 
-func (w *Worker) setDrained(isDrained bool) {
-	w.drainedLock.Lock()
-	defer w.drainedLock.Unlock()
-
-	w.isDrained.Store(isDrained)
+func (w *Worker) Continue() error {
+	if err := w.runtime.Continue(); err != nil {
+		return err
```

Comment:
`errors.Wrap(err, ...)`

Reply:
@liranbg we do the same `return err` in `Drain` and `Terminate`. Err will be wrapped on the next step. We have 1-1 relationships between worker and runtime, so maybe we can leave as is?